operations/user/usersOperation.js

```diff
@@ -0,0 +1,198 @@
+				expiresIn: "1d"
+			});
+			await this.userService.userServices.updateUser({_id: user._id},{accessToken})
+			return this.emit(SUCCESS, {
+				accessToken,
+				firstname: user.firstname,
+				lastname: user.lastname,
+				email: user.email,
+				role: user.role,
+				photoName :user.photoName
+			})
+
+		} catch (error) {
+			console.log(error)
+			if (error.message === 'ValidationError') {
+				return this.emit(VALIDATION_ERROR, error);
+			} else if (error.message === "NotFoundError") {
+				return this.emit(NOT_FOUND, error)
+			} else {
+				return this.emit(ERROR, error);
+			}
+		}
+	}
+
+	async getUser(query,token) {
+		const {
+			SUCCESS,
+			ERROR,
+			VALIDATION_ERROR,
+			NOT_FOUND
+		} = this.outputs;
+		try {
+			
+			
+			if(user.role === 'onboarder')
+			{
+			const users = await this.userService.userServices.getUsers(query)
+			return this.emit(SUCCESS, {
+				users
+			})}
+			else
+			return this.emit(SUCCESS, {
+				message:'you are not an onboarder'
+			})
+		} catch (error) {
+			console.log(error)
+			if (error.message === 'ValidationError') {
+				return this.emit(VALIDATION_ERROR, error);
+			} else if (error.message === "NotFoundError") {
+				return this.emit(NOT_FOUND, error)
+			} else {
+				return this.emit(ERROR, error);
+			}
+		}
+	}
+	async addUser(data,token,photoName) {
+		const {
+			SUCCESS,
+			ERROR,
+			VALIDATION_ERROR,
+			NOT_FOUND
+		} = this.outputs;
+		try {
+			const tokenget = token.split(' ')[1];
+			const user = await this.userService.userServices.findOneUser({accessToken : tokenget })
+
+			if(user.role === 'onboarder')
+			{
+				
+				data.photoName = photoName;
+				console.log(photoName);
+			const user = await this.userService.userServices.addUser(data)
+			return this.emit(SUCCESS, {
+				user
+			})}
+			else 
+			return this.emit(SUCCESS, {
+				message:'you are not an onboarder'
+			})
+		} catch (error) {
+			console.log(error)
+			if (error.message === 'ValidationError') {
+				return this.emit(VALIDATION_ERROR, error);
+			} else if (error.message === "NotFoundError") {
+				return this.emit(NOT_FOUND, error)
+			} else {
+				return this.emit(ERROR, error);
+			}
+		}
+	}
+	async updateProfile(data,token) {
+		const {
+			SUCCESS,
+			ERROR,
+			VALIDATION_ERROR,
+			NOT_FOUND
+		} = this.outputs;
+		try {
+			const tokenget = token.split(' ')[1];
+			const user = await this.userService.userServices.findOneUser({accessToken : tokenget })
+			await this.userService.userServices.updateUser(user._id,data)
+			const updatedUser = await userService.userServices.findOneUser(user._id)
+			return this.emit(SUCCESS, {
+				updatedUser
+			})
+			
+		} catch (error) {
+			console.log(error)
+			if (error.message === 'ValidationError') {
+				return this.emit(VALIDATION_ERROR, error);
+			} else if (error.message === "NotFoundError") {
+				return this.emit(NOT_FOUND, error)
+			} else {
+				return this.emit(ERROR, error);
+			}
+		}
+	}
+	async updateUser(data,token) {
```

Comment:
Thea data object should be an object parameter with specifically what do you need from req.body